src/Nethermind/Nethermind.State/IWorldState.cs

```diff
@@ -17,6 +17,10 @@ namespace Nethermind.State;
 /// </summary>
 public interface IWorldState : IJournal<Snapshot>, IReadOnlyStateProvider
 {
+
+    public IDisposable BeginScope(Hash256 stateRoot);
```

Comment:
Maybe `BlockHeader? parent` would be better? Could support world state that does not calculate state root.

Reply:
yes, that is the idea
this is just a try to see if things work with stateRoot - easy to do in tests - and then i can modify it to take blockHeader+spec instead of stateRoot.